command_line/symmetry.py

```diff
@@ -224,6 +225,39 @@ def eliminate_sys_absent(experiments, reflections):
     return reflections
 
 
+def get_subset_for_symmetry(params, experiments, reflection_tables):
+    """Select an image range for symmetry analysis, or just select
+    the first 360 degrees of data."""
+    refls_for_sym = []
+    if params.exclude_images:
+        experiments = exclude_image_ranges_from_scans(
+            reflection_tables, experiments, params.exclude_images
+        )
+        for refl, exp in zip(reflection_tables, experiments):
+            sel = get_selection_for_valid_image_ranges(refl, exp)
+            refls_for_sym.append(refl.select(sel))
+    else:
+        # use first 360 degrees if <360 deg i.e. first measured data.
+        for expt, refl in zip(experiments, reflection_tables):
```

Comment:
I think the following would achieve the same results in a lot fewer lines of code - at least with these changes the tests still pass:
```
        for expt, refl in zip(experiments, reflection_tables):
            scan_end = int(math.ceil(360/abs(expt.scan.get_oscillation()[1])))
            if scan_end < len(expt.scan):
                refl = refl.select(refl["xyzobs.px.value"].parts()[2] <= scan_end)
            refls_for_sym.append(refl)
```

Reply:
Yes I think I agree, the px values always go from 0 upwards regardless of scan direction, so this always selects the first measured 360 degrees? Good spot.